ksql-common/src/test/java/io/confluent/ksql/util/KsqlConfigTest.java

```diff
@@ -415,4 +415,46 @@ public void shouldNotListUnresolvedServerConfig() {
     // Then:
     assertThat(result.get("some.random.property"), is(nullValue()));
   }
+
+  @Test
+  public void shouldDefaultOptimizationsToOn() {
+    // When:
+    final KsqlConfig config = new KsqlConfig(Collections.emptyMap());
+
+    // Then:
+    assertThat(
+        config.getKsqlStreamConfigProps().get(StreamsConfig.TOPOLOGY_OPTIMIZATION),
+        equalTo(StreamsConfig.OPTIMIZE));
+  }
+
+  @Test
+  public void shouldDefaultOptimizationsToOffForOldConfigs() {
+    // When:
+    final KsqlConfig config = new KsqlConfig(false, Collections.emptyMap());
```

Comment:
just a thought... but boolean params can be a bit naff when it comes to code readability, e.g. what's the difference between these two lines?

```
new KsqlConfig(true, Collections.emptyMap());
new KsqlConfig(false, Collections.emptyMap());
```

If  you don't know the code you need to go check what that boolean means when groking the code.

There are two common patterns for improving readability:

1. Use an enum:

```
new KsqlConfig(CURRENT_CONFIG, Collections.emptyMap());
new KsqlConfig(LEGACY_CONFIG, Collections.emptyMap());
```

2. Use a factory:
```
KsqlConfig.currentConfig(Collections.emptyMap());
KsqlConfig.legacyConfig(Collections.emptyMap());
```

I realise you haven't introduced this boolean in this review, but ever time I see it I have to go check the code to work out what it means.

A worthwhile change?

Reply:
Yeah I totally agree here. I'll change it to an enum.